lua/swenv/api.lua

```diff
@@ -139,6 +139,24 @@ local get_micromamba_base_path = function()
   end
 end
 
+local get_poetry_venvs_base_path = function()
+  local which_poetry = io.popen('which poetry')
```

Comment:
I'm starting to get a bit worried about performance when all these functions are called and some call out to external executables. I don't use it but some users use the `auto_venv` which calls this on `FileType`. I wonder if this should be done async. This is somewhat independent from this PR but something I thought of now.

I also wonder, the current implementation supports the `venvs_path` setting. Would it be an option to support that being a function and instead supply this as a optional utility function which a user can use instead of the default `venvs_path`?

Reply:
Whell, i will replace sys calls this with parametrs in plugin config.

Reply:
I realized that poetry support does not require any code mutation. I can just use venvs_path.
But I'm still thinking. which is useful to replace venvs_path with venvs_paths, with one, two or many directories.

Reply:
Also i can set use "has_conda" 'has_pixi', 'has_micromama' param. If it set to false - functions will not be called.
Like this:
```lua
  if settings.spec_envs == nil then
    return venvs
  end
  if 'conda' in setting.spec_envs then
    vim.list_extend(venvs, get_venvs_for(get_conda_base_path(), 'conda'))
    vim.list_extend(venvs, get_conda_base_env())
  end
```
This will speed up the plugin a bit, but... at the cost of backward compatibility.
It may be worth making a separate branch for such a change. For example, 'nightly_speedup'.

Reply:
In the meantime, I agree - everything except the usual venus should be handled asynchronously.

Reply:
> But I'm still thinking. which is useful to replace venvs_path with venvs_paths, with one, two or many directories.

Yeah I think that would be useful, maybe just support the setting to be string, table or function returning string or table? This could then be backwards compatible.

> Also i can set use "has_conda" 'has_pixi', 'has_micromama' param. If it set to false - functions will not be called.
Like this:

I think that's a good idea, maybe it can default to `true` for now to be backwards compatible and then at least it can be disabled if speed is an issue?